src/editors/containers/VideoGallery/hooks.js

```diff
@@ -0,0 +1,194 @@
+  const [
+    showSelectVideoError,
+    setShowSelectVideoError,
+  ] = module.state.showSelectVideoError(false);
+  const [
+    showSizeError,
+    setShowSizeError,
+  ] = module.state.showSizeError(false);
+  const filteredList = module.filterList({ ...searchSortProps, videos });
+  return {
+    galleryError: {
+      show: showSelectVideoError,
+      set: () => setShowSelectVideoError(true),
+      dismiss: () => setShowSelectVideoError(false),
+      message: messages.selectVideoError,
+    },
+    // TODO We need to update this message when implementing the video upload screen
+    inputError: {
+      show: showSizeError,
+      set: () => setShowSizeError(true),
+      dismiss: () => setShowSelectVideoError(false),
+      message: messages.fileSizeError,
+    },
+    galleryProps: {
+      galleryIsEmpty: Object.keys(filteredList).length === 0,
+      searchIsEmpty: filteredList.length === 0,
+      displayList: filteredList,
+      highlighted,
+      onHighlightChange: (e) => setHighlighted(e.target.value),
+      emptyGalleryLabel: messages.emptyGalleryLabel,
+      showIdsOnCards: true,
+      height: '100%',
+    },
+    selectBtnProps: {
+      onclick: () => {
+        // TODO Update this when implementing the selection feature
+      },
+    },
+  };
+};
+
+export const fileInputHooks = () => {
+  // TODO [Update video] Implement this
+  const ref = React.useRef();
+  const click = () => ref.current.click();
+
+  return {
+    click,
+    addFile: () => {},
+    ref,
+  };
+};
+
+export const buildVideos = ({ rawVideos }) => {
+  let videos = [];
+  const videoList = Object.values(rawVideos);
+  if (videoList.length > 0) {
+    videos = videoList.map(video => ({
+      id: video.edx_video_id,
+      displayName: video.client_video_id,
+      externalUrl: video.course_video_image_url,
+      dateAdded: video.created,
+      locked: false,
+      thumbnail: video.course_video_image_url,
+      status: video.status,
+      statusBadgeVariant: module.getstatusBadgeVariant({ status: video.status }),
+      duration: video.duration,
+      transcripts: video.transcripts,
+    }));
+  }
+  return videos;
+};
+
+export const getstatusBadgeVariant = ({ status }) => {
+  switch (status) {
+    case filterKeys.failed:
+      return 'danger';
+    case filterKeys.uploading:
+    case filterKeys.processing:
+      return 'light';
+    default:
+      return null;
+  }
+};
+
+export const videoHooks = ({ videos }) => {
+  const searchSortProps = module.searchAndSortHooks();
+  const videoList = module.videoListHooks({ searchSortProps, videos });
+  const {
+    galleryError,
+    galleryProps,
+    inputError,
+    selectBtnProps,
+  } = videoList;
+  const fileInput = module.fileInputHooks();
+
+  return {
+    galleryError,
+    inputError,
+    fileInput,
+    galleryProps,
+    searchSortProps,
+    selectBtnProps,
+  };
+};
+
+export default {
+  videoHooks,
+  buildVideos,
+};
```

Comment:
None of the functions in this file seem to be hooks? Perhaps the file is misnamed?

Reply:
Updated :+1: